### ISSUE TITLE
phantomjs alternative

### ISSUE THREAD

**Issue by wioux** [EFForg/phantom-of-the-capitol#124]
Should we find an alternative to phantomjs? The maintainer has stepped down.

**Comment by mfb** [comment]
There is now firefox headless https://mykzilla.org/2017/08/30/headless-firefox-in-node-js-with-selenium-webdriver/ or I guess more popularly, chrome headless.

**Comment by j-ro** [comment]
We'd support a move. I heard chrome headless is much faster. That said, we have no development time to devote to this at the moment unfortunately. And I don't feel a ton of urgency about it, we don't even keep up with phantom updates as it is.

**Comment by wioux** [comment]
Do we still need to support webkit/waitir? `REQUIRES_WAITIR` is empty and all the bioguide ids from `REQUIRES_WEBKIT` are house members so we can clear that out, but I'm not sure what the need for the alternative drivers was originally and whether it might come up again. We could really simplify parts of the app if we removed support for those drivers.

**Comment by j-ro** [comment]
I think that's probably fine over here, yeah...

**Comment by koops76** [comment]
https://github.com/GoogleChrome/puppeteer

**Comment by j-ro** [comment]
has anyone started work on this?

**Comment by wioux** [comment]
Not yet @j-ro.

**Comment by j-ro** [comment]
Thanks @wioux, us either, though it's starting to become more important for us. I'll let you know if it lands on my roadmap. Can you do the same, so we don't duplicate work?

**Comment by wioux** [comment]
Definitely, I'll let you know.

**Comment by j-ro** [comment]
We're actually doing a bit of initial investigation work on this today, maybe tomorrow too. We'll let you know how it works. There may be just a drop-in replacement that works with capybera, if so, will be fairly easy....

**Comment by j-ro** [comment]
Update here -- we have chromedriver running, but it's probably not quite ready for prime time. It works, but seeing some hard to debug timeout errors, and it's missing some features like blacklists. We're going to run it as an optional switch for certain yamls since it helps in some cases, but we're not going to entirely switch. If there's large appetitive for the code we can put together a PR, but it's very much a WIP.

**Comment by k-stewart** [comment]
Hey @j-ro, this is becoming more important for us.  Have you found a solution you like?

**Comment by j-ro** [comment]
No, we're still with phantom. Chromedriver works but not as consistently, and it doesn't have many hooks and options to debug and tune. We haven't looked at it since January, maybe that's changed, but we're not planning a switch.

**Comment by ghost** [comment]
I'll chime in with my experience as I have worked with  puppeteer, and phantomjs, and various [selenium webdriver](https://github.com/SeleniumHQ/selenium) implementations like chromedriver and   [geckodriver](https://github.com/mozilla/geckodriver#geckodriver).
Puppeteer provides a high level API that is quite easy to work with for basic scraping. They publish [extensive documentation](https://github.com/GoogleChrome/puppeteer/blob/master/docs/api.md) as well. If needing to get something done quick, I think this is a strong contender. It is a JavaScript only API as far as I know.
Selenium webdriver implementations give you more flexibility with the browser you run the automation in but require more programming and configuration to get working. The API is also implemented in different programming languages. [Firefox's headless documentation](https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Headless_mode#Automated_testing_with_headless_mode) also recommends using selenium webdriver for testing automation.

**Comment by ghost** [comment]
Just discovered @k-stewart 's work in #141 as well.

**Comment by woopsn** [comment]
Hi @efx. Our contact-congress work has moved over to [EFForg/congress_forms_api](https://github.com/EFForg/congress_forms_api) to fix this and other issues. Sorry we didn't properly archive this repo -- I'm going to do that now.

**Comment by ghost** [comment]
Thanks @wioux. I had found this repository from EFF's homepage, so we should probably update those link(s) as well.

**Comment by danielmroberts** [comment]
> Hi @efx. Our contact-congress work has moved over to [EFForg/congress_forms_api](https://github.com/EFForg/congress_forms_api) to fix this and other issues. Sorry we didn't properly archive this repo -- I'm going to do that now.

This repo is still not archived. We were about to roll out a system we have been working on for a while based on phantom of the capitol before noticing your comment :(
